### ISSUE TITLE
`read` fails to properly read variants

### ISSUE THREAD

**Issue by bklaric** [rowtype-yoga/purescript-yoga-json#23]
Please see this [minimal reproducible example on Try PureScript](https://try.purescript.org/?code=LYewJgrgNgpgBAWQIYEsB2cDuALGAnGAKEJWAAcQ8AXOABQKgjCJPMpoBEkqkA6AMRBQwSAEaw4ACgBmQsAEpWFanACi06TADGNSes07FpZTQAqeAJ60IBAMpa8KMruwBGADRxsAJk9lPVDAAHlSeUCgAzqFw6Mxo0eFoANaeBGjMeJ5a4DBGbCoAmiAA5nwAUrYA8gBySuxwXDy8AGpIjkjxdSr62lS8AMIgaBFCRMb1phZkMLz0IEEWXTQAqsNImgMg%2BFpj%2BZzcfMgWoixUU-D8SNBnCDAREUjF8AC8cK3tnXBfUnBkjzAAGRASGYYDgAC5wXAAN4AX0I3zgnhEPEGUFgOhgYMhMJGwBgjSQEKhtiojjQxXhiMUhDO0zggzwBB0t3u-zgr3eKA6VAR30kcBg5DOxJhVO%2BnkwKCo2EJouheIJB1FpPJlL5XxpwFQGBxPR0cFW0sI2vQHLgsF5X0kfyeQJBWNFRxOcEu1wsrIeT3k5oIIIA%2BnAACRYRyBYNSdAAKykcwWosDACJbYDgaDEz64aK3VAbncvbk%2BZIUUg0RjAtioc6Lldcx78-8fa8-WBAyHMGH4CHJNHY3h5hYE3BEyWy70sRncSB8YSoYmkFsIInYdna3m2d6i0LnIOcdWGZRmVRPY3fTAAxGO9Ku5G0DHJHHd1Ck9uzpOszjGUeT5vrVKZXKe5IMc8Bfr0P65GeF7tp2EY9nefYDkOib-rKBzvoqs7DguMBLiun6HuBDa-jEGBgCAfJQCUw6mLgcCyOiIBShScBUJgIBwAAbm03LxBEcARNgIDQGCLpgCgGj4DA8SeKIEA0DKMCDm08CKQJSD4rwiaUSUthCZgvz-PaoI6cUelMXAo5COWWKmTRdEMVRzHFKx7FcTxPL8SplkSZoaTREg-GKYOgnCcIcAnFppnmQZr6LF8VFmfpWDSmhPBAA).

Basically, once I `write` a variant value, `read`-ing it back sometimes results in `read` picking up the wrong variant case. To make things worse, this seems to happen only with some combinations of label names.

![image](https://github.com/rowtype-yoga/purescript-yoga-json/assets/9405688/55535a3a-5ead-4f63-8105-17711e69051d)

**Comment by i-am-the-slime** [comment]
Thank you for this report, I haven't seen it for almost a year, very bad.

**Comment by i-am-the-slime** [comment]
I wonder if this is because it's possible to construct a `{}` from a `{ someData: xyz }` and variants get tried in alphabetic order?
